### PR TITLE
chore: refactor models

### LINKED ISSUE #35: Refactor core logic, models and traits
# Issue | Checkers

## 1. Type of Issue

- [ ] Bug  
- [ ] Feature Request  
- [x] Enhancement  
- [ ] Documentation  
- [ ] Other  

---

## 2. Brief Description  

Description
To improve code maintainability, scalability, and readability, we propose a refactor of the current implementation to adopt a more modular structure. The primary focus will be to:

1- Migrate logic related to specific models into their respective traits
2-Restructure models into separate directories

Example Refactor

```
#[generate_trait]

 impl PieceImpl of PieceTrait {
         fn new(
             session_id: u64,
             row: u8,
             col: u8,
             player: ContractAddress,
             position: Position,
         ) -> Piece {
                   Piece {
                       session_id,
                       row,
                       col,
                       player,
                       position,
                       is_king: false,
                       is_alive: true,
                   }
         }
```

---

## 3. Current Behavior  
  

---

## 4. Expected Behavior  
  

---

## 5. Steps to Reproduce (For Bugs)

1. Step 1  
2. Step 2  
3. Step 3  

---

## 6. Evidence (Optional)  

---

## 7. Additional Context (Optional)  

---

### PR DESCRIPTION
**Pull request by RajeshRk18**
Closes #35

### CODE REVIEW

**Review by Gianfranco99 — CHANGES_REQUESTED** [review]
Test and build don't work, do sozo build and sozo test, they should all pass green
@RajeshRk18 
Screenshot 2024-12-26 090517

**Comment by RajeshRk18** [comment]
> Test and build don't work, do sozo build and sozo test, they should all pass green
> 
> @RajeshRk18 
> 
> Screenshot 2024-12-26 090517

Sure! I will resolve this asap. I didn't get any of these errors

**Comment by gianalarcon** [comment]
Hi @RajeshRk18. Any updates on this?

**Comment by RajeshRk18** [comment]
I will push the changes on Monday.

**Comment by RajeshRk18** [comment]
@gianalarcon @Gianfranco99 Tests are passing now!

image

**Comment by gianalarcon** [comment]
Hi @RajeshRk18 sorry, we just updated develop branch, pls fix conflicts

**Comment by RajeshRk18** [comment]
@gianalarcon Fixed!

File: dojo-starter/src/tests/test_world.cairo [hunk 1]
```diff
@@ -11,16 +11,16 @@ mod tests {
...
+    use dojo_starter::models::position::Position;
+    use dojo_starter::models::session::{Session, SessionTrait, SessionImpl, m_Session};
+    use dojo_starter::models::player::{Player, PlayerTrait, PlayerImpl, m_Player};
+    use dojo_starter::models::counter::{Counter, CounterTrait, m_Counter};
 
     fn namespace_def() -> NamespaceDef {
         let ndef = NamespaceDef {
-            namespace: "checkers_marq",
-            resources: [
+            namespace: "checkers_marq", resources: [
```
Comment (847850277) [inline dojo-starter/src/tests/test_world.cairo]: please check fmt using `scarb fmt --check`

File: dojo-starter/src/tests/test_world.cairo [hunk 2]
```diff
@@ -41,8 +40,7 @@ mod tests {
         [
             ContractDefTrait::new(@"checkers_marq", @"actions")
```
Comment (847850277) [inline dojo-starter/src/tests/test_world.cairo]: please check fmt using `scarb fmt --check`

**Comment by gianalarcon** [comment]
workflow for code format should work same as this [PR](https://github.com/The-Marquis-Gaming/checkers-dojo/pull/44)

**Comment by RajeshRk18** [comment]
@gianalarcon How will it be paid sir?

**Comment by gianalarcon** [comment]
Thank you so much for your contribution. I will submit your reward to onlydust platform on few hours

**Pull request merged**